### DISCUSSION TITLE
DOI and DOI metadata versioning

### DISCUSSION THREAD

**Discussion by NRayya** [NFDI4Chem/nmrxiv#1169] [Features]
In nmrXiv we already have basic DOI metadata versioning, however, we want to go to the next level of versioning where we would like to distinguish between two concepts:

- DOI versioning: When the data itself (not the metadata) change, then this is a new version of the data and a new doi should be provided. Example: https://doi.org/10.57992/nmrxiv.p1 become https://doi.org/10.57992/nmrxiv.p1.v2
- DOI **metadata** versioning:  When the metadata change, then there is no new version of the data. The doi doesn't change but the metadata version changes. This can be visible in the datacite export
![image](https://github.com/user-attachments/assets/8b2a1b58-bebb-4c3e-a9ff-d624df30fd5d)

What we would like to discuss: 

1. What is considered data? In nmrXiv we would like to consider the **files** as the data and when the user adds, replaces, or removes files, then a new doi version is needed.
2. What is considered metadata? In nmrXiv we would like to consider everything provided as user entry in fields or exported from the files as metadata regardless of being domain independent or specific metadata.
3. There is an approach (followed by Zenodo) employs generic dois (without version suffix) which lead to the last version while specific versions have "v" as a suffix. 
  - The problem with implementing this approach is that so far we have only be providing dois without versioning (without v suffix) and people have already been citing these dois indicating the first version. If we change the meaning and redirecting of dois it will affect these old citations.
4. How to handle new versions on the level of database? 
5. How to handle the different versions of metadata at the level of database? 
6. What should the user have access to? separate views of the different doi versions? What metadata should these versions show? only the last one corresponding to the doi version? Should the user have download access of datacite metadata with the old versions?

**Comment by tilfischer** [comment]
Dear Noura,

Thank you for pointing me on this discussion. I agree with what is written above the screenshot (change to data -> new DOI, change to metadata -> update (DOI) metadata). 

1.  I agree and would like to add that we also need to consider analysis data e.g. peaks, integrals, ranges/multiplets and assignments. It could happen that this data changes because it is added or there is a correction. Is this a change to the data? I would say so. However, this is not a change to the data uploaded. As far as I know, adding analysis data in nmrXiv via NMRrium does not (and should not) alter the original data uploaded. Out of curiosity: How do you currently provide that additionally added analysis data if users download data? Here, we are possible back in the file format standards schlamassel. ;)
2. You are right. Metadata can be domain-independent or domain-specific. It could be even subdiscipline-specific.
3. The approach by Zenodo sound reasonable. Currently, there are not that many users. We could still inform them about the change via email, so that they may update the links. If they cannot update the link (e.g. because the link is in a DAS of an article, which cannot be changed anymore) than the DOI leads the the most recent version, which is also a'right. Reverse: I could also imagine situation where users access data in nmrXiv via a DOI provided in a DAS. They are subsequently lead to the original dataset, reuse the data and later (may) found out that there is a new version they missed, hence, it is alright that https://doi.org/10.57992/nmrxiv.p1 always leads to the newest version. The https://doi.org/10.57992/nmrxiv.p1.v1 can be registered, if users update the data and https://doi.org/10.57992/nmrxiv.p1.v2 is registered.
4. This is out of my scope.
5. Also out of my scope. 
6. We need dataset and dataset metadata versioning. User should be able to access the the data of each version. As each new version of a dataset has a new DOI, there is separate metadata which users can access. They should also be able to see if there were changes of the metadata (for one isolated DOI or for each version-DOI), as this is the case for RADAR (for the isolated DOIs). I believe that users should also be able to download older version.

Best,
Tillmann
